### PR TITLE
Allow all flags to be repeated (false by default).

### PR DESCRIPTION
**Pull request by Tener**
This makes the Kingpin compatible with the UNIX convention and makes it much easier to compose commands and build command aliases.

Before change:
```
➜  ~ tsh --debug --debug login
...
ERROR: flag 'debug' cannot be repeated
```

After this change (and enabling this option via `Application.AllRepeatable(true)`) the above works without problems.

### CODE REVIEW

**Review by smallinsky — APPROVED** [review]

File: app_test.go [hunk 1]
```diff
@@ -79,6 +78,12 @@ func TestRepeatableFlags(t *testing.T) {
 	assert.Error(t, err)
 	_, err = c.Parse([]string{"--b=foo", "--b=bar"})
 	assert.NoError(t, err)
+
+	c.AllRepeatable(true)
+	_, err = c.Parse([]string{"--a=foo", "--a=bar"})
+	assert.NoError(t, err)
```
Comment (rosstimothy) [inline app_test.go:84]: Add an assertion for the value of `a` in this case?

**Review by rosstimothy — APPROVED** [review]
What happens in the event a flag isn't a `repeatableFlag` but `allRepeatable == true`? The last flag provided is used? I'm assuming that is what UNIX convention means, but could you perhaps make the comments a bit more explicit?

**Comment by Tener** [comment]
> What happens in the event a flag isn't a `repeatableFlag` but `allRepeatable == true`? The last flag provided is used? I'm assuming that is what UNIX convention means, but could you perhaps make the comments a bit more explicit?

What technically happens is that the flag setter is called multiple times. Kingpin doesn't actually care what the setter does, but in practice:
- for slice-valued flags, the values will be accumulated
- for single-value flags, the last call will be effective

The first is old news; the second one is enabled by this PR.

File: app_test.go (hunk 1 above)
Comment (Tener) [reply app_test.go:84]: Done, thanks.

**Pull request merged**